### PR TITLE
Integrating ppom update tool into mobile build

### PR DESCRIPTION
**Pull request by jpuri**
## **Description**
Add ppom-view tool https://github.com/MetaMask/ppom-view. into metamask mobile to add generation `ppom.html.js` generation into mobile build.

## **Manual testing steps**

_1. Run mobile build
_2. Build should run smoothly
_3. Build should generate new `ppom.html.js`

## **Related issues**

_Fixes: MetaMask/MetaMask-planning#1362

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained:
  - [x] What problem this PR is solving.
  - [x] How this problem was solved.
  - [x] How reviewers can test my changes.
- [x] I’ve indicated what issue this PR is linked to: Fixes #???
- [ ] I’ve included tests if applicable.
- [x] I’ve documented any added code.
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by jpuri** [comment]
@SocketSecurity ignore-all

File: ppom/webpack.config.html-js.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+const path = require('path');
+
+module.exports = {
+  entry: './src/ppom.html.js',
+  output: {
+    path: path.resolve(__dirname, '../app/lib/ppom'),
```
Comment (naugtur) [inline ppom/webpack.config.html-js.js:6]: IMHO This says the app/lib/ppom folder should not be committed to git but gitignored. 

To avoid errors where the file is missing or outdated the build for ppom could become an early build step of the app itself.

File: scripts/postinstall.sh [hunk 2]
```diff
@@ -1,3 +1,5 @@
+source ./scripts/build_ppom.sh
```
Comment (leotm) [inline scripts/postinstall.sh:1]: could be worth labelling this `echo "1. Build ppom"` for visibility
then moving it below and renaming the steps

Comment (leotm) [reply scripts/postinstall.sh:1]: or moving to package.json > scripts
`"setup": "yarn clean && yarn allow-scripts && ./scripts/build_ppom.sh && ./scripts/postinstall.sh",`

Comment (jpuri) [reply scripts/postinstall.sh]: Thanks @leotm , I made this change

File: ppom/webpack.config.html-js.js (hunk 1 above)
Comment (jpuri) [reply ppom/webpack.config.html-js.js:6]: PR is updated, `app/lib/ppom/ppom.html.js` is added to `.gitignore`.

**Comment by leotm** [comment]
looks like couple bits in CI left to fix then we're good 2 go

**Review by leotm — COMMENTED** [review]
looks like couple bits in CI left to fix then we're good 2 go

File: ppom/webpack.config.html-js.js (hunk 1 above)
Comment (jpuri) [reply ppom/webpack.config.html-js.js:6]: I found an issue with ignoring this file. If I do not commit it, I need to add it to other commands for unit tests, e2e tests, etc.

**Comment by seaona** [comment]
I am able to build the app successfully @jpuri . While QA is taking place, I'll add the findings in this same comment:

### 1. `Request may not be safe` not shown
Whenever there is an issue with ppom validating the transction (i.e. the account does not have enough funds) instead of seeing the `Request may not be safe` warning, we don't see any warning.

https://github.com/MetaMask/metamask-mobile/assets/54408225/73c1d894-06f2-44ff-9488-f673a1d5fbc4
**Repro:**
1. Enable Blockaid
2. Go to the test dapp
3. Select an empty account
4. Trigger the Malicious ERC20 transfer or Malicious ERC20 Approve transfer
5. We should see the `Request may not be safe` warning



### 2. See details arrow icon missmatch between platforms
Whenever details are hidden, we can see how the icon arrow is looking up, instead of looking down (see comparison with Extension). Not sure which one is correct and which one is wrong cc @bschorchit 

image

**Comment by seaona** [comment]
Since the feature is behind a feature flag, we decided to open tickets for solving the issues in separate PRs cc @jpuri

**Review by segun — DISMISSED** [review]

**Comment by jpuri** [comment]
Bitrise: https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/da0fff27-1cf9-4536-871f-143e233f64bd

**Review by blackdevelopa — DISMISSED** [review]

File: .eslintrc.js [hunk 3]
```diff
@@ -98,7 +98,10 @@ module.exports = {
     'import/no-amd': 2,
     'import/no-commonjs': 2,
     'import/no-duplicates': 2,
-    'import/no-extraneous-dependencies': 2,
+    'import/no-extraneous-dependencies': [
+      'error',
+      { packageDir: ['./ppom', './'] },
+    ],
```
Comment (leotm) [inline .eslintrc.js:104]: is this change still needed or can we revert it?

now with `/ppom` being ignored in our root `.eslintignore`

Comment (leotm) [reply .eslintrc.js:104]: (noting also `ppom/.eslintignore` is ignoring `/webpack.config.html-js.js` and `/webpack.config.html.js`)

File: ppom/README.md [hunk 4]
```diff
@@ -0,0 +1,3 @@
+# ppom-view
+
+This is tool consumes Blockaid's [PPOM package](https://www.npmjs.com/package/@blockaid/ppom_release) and packages it with gluing code to generate `ppom.html.js`. `ppom.html.js` is embedded into webview.
```
Comment (leotm) [inline ppom/README.md:3]:
```suggestion
This is tool consumes Blockaid's [PPOM package](https://www.npmjs.com/package/@blockaid/ppom_release) and packages it with gluing code to generate `ppom.html.js` and `ppom.html.js` that are both embedded into a webview.
```

File: .eslintrc.js (hunk 3 above)
Comment (jpuri) [reply .eslintrc.js]: Thanks @leotm , that is good point.

I made the change to remove `./ppom` above.

**Review by segun — DISMISSED** [review]

**Review by segun — DISMISSED** [review]

**Comment by blackdevelopa** [comment]
It seems we have some conflicts on this PR.

**Review by blackdevelopa — APPROVED** [review]

**Review by segun — APPROVED** [review]

**Comment by jpuri** [comment]
Bitrise: https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/954cfa18-6969-46a4-a20f-631e341a0799
